lib/matplotlib/animation.py

```diff
@@ -971,17 +971,18 @@ def __init__(self, fig, artists, *args, **kwargs):
 
     def _init_draw(self):
         # Make all the artists involved in *any* frame invisible
-        axes = []
+        figs = set()
```

Comment:
Good catch on that and because i bet it is defaulting to using the object id

On Wed, Aug 12, 2015, 11:27 AM Benjamin Root <notifications@github.com>
wrote:

> In lib/matplotlib/animation.py
> <https://github.com/matplotlib/matplotlib/pull/4738#discussion_r36873399>:
>
> > @@ -971,17 +971,18 @@ def __init__(self, fig, artists, *args, **kwargs):
> >
> >      def _init_draw(self):
> >          # Make all the artists involved in *any* frame invisible
> > -        axes = []
> > +        figs = set()
>
> wait a minute... figures are mutable. How is it that we can have a set of
> them?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/matplotlib/matplotlib/pull/4738/files#r36873399>.
>